src/platform/silabs/rs911x/BLEManagerImpl.cpp

```diff
@@ -1101,6 +1117,36 @@ void BLEManagerImpl::StartBleAdvTimeoutTimer(uint32_t aTimeoutInMs)
     }
 }
 
+
+void BLEManagerImpl::BleSendIndicationTimeoutHandler(TimerHandle_t xTimer)
+{
+        sInstance.HandleSoftTimerEvent();
+}
+
+void BLEManagerImpl::CancelBleSendIndicationTimeoutTimer(void)
+{
+    if (xTimerStop(sbleSendIndicationTimeoutTimer, pdMS_TO_TICKS(0)) == pdFAIL)
+    {
+        ChipLogError(DeviceLayer, "Failed to stop BledAdv timeout timer");
+    }
+}
+
+void BLEManagerImpl::StartBleSendIndicationTimeoutTimer(uint32_t aTimeoutInMs)
+{
+    if (xTimerIsTimerActive(sbleSendIndicationTimeoutTimer))
+    {
+        CancelBleAdvTimeoutTimer();
```

Comment:
Is this the right timer to be cancelling here? If so, it would help to have a comment explaining what's going on.

Reply:
I agree with Sergei; this doesn't seem to be the right timer...